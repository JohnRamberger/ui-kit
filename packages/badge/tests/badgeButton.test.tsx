import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import { BadgeButton } from "../";

const StringComponent = (): JSX.Element => {
  return <span>string</span>;
};
const fn = () => null;

describe("BadgeButton", () => {
  it("default", () => {
    const component = mount(<BadgeButton onClick={fn}>default</BadgeButton>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("success", () => {
    const component = mount(
      <BadgeButton onClick={fn} appearance="success">
        success
      </BadgeButton>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it("primary", () => {
    const component = mount(
      <BadgeButton onClick={fn} appearance="primary">
        primary
      </BadgeButton>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it("danger", () => {
    const component = mount(
      <BadgeButton onClick={fn} appearance="danger">
        danger
      </BadgeButton>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it("warning", () => {
    const component = mount(
      <BadgeButton onClick={fn} appearance="warning">
        warning
      </BadgeButton>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it("outline", () => {
    const component = mount(
      <BadgeButton onClick={fn} appearance="outline">
        outline
      </BadgeButton>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it("accept jsx as children", () => {
    const component = mount(
      <BadgeButton onClick={fn}>
        <StringComponent />
      </BadgeButton>
    );

    expect(toJson(component)).toMatchSnapshot();
  });

  it("contains the right tabindex", () => {
    const wrapper = mount(
      <BadgeButton onClick={jest.fn()} tabIndex={-10}>
        default
      </BadgeButton>
    );
    const element = wrapper.find("span").props();
    expect(element.tabIndex).toBe(-10);
  });

  it("triggers onClick on click", () => {
    const onBadgeButtonClick = jest.fn();
    const wrapper = mount(
      <BadgeButton onClick={onBadgeButtonClick}>default</BadgeButton>
    );
    wrapper.simulate("click");
    expect(onBadgeButtonClick).toHaveBeenCalled();
  });
});
