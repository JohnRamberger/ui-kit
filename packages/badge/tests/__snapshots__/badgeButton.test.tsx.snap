// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeButton accept jsx as children 1`] = `
<Memo(BadgeButton)
  onClick={[Function]}
>
  <Clickable
    action={[Function]}
    data-cy="badgeButton"
    disableFocusOutline={false}
    role="button"
    tabIndex={-1}
  >
    <span
      className="css-1vry5xk"
      data-cy="badgeButton"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={-1}
    >
      <StringComponent>
        <span>
          string
        </span>
      </StringComponent>
    </span>
  </Clickable>
</Memo(BadgeButton)>
`;

exports[`BadgeButton danger 1`] = `
<Memo(BadgeButton)
  appearance="danger"
  onClick={[Function]}
>
  <Clickable
    action={[Function]}
    data-cy="badgeButton"
    disableFocusOutline={false}
    role="button"
    tabIndex={-1}
  >
    <span
      className="css-1h0zebo"
      data-cy="badgeButton"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={-1}
    >
      danger
    </span>
  </Clickable>
</Memo(BadgeButton)>
`;

exports[`BadgeButton default 1`] = `
<Memo(BadgeButton)
  onClick={[Function]}
>
  <Clickable
    action={[Function]}
    data-cy="badgeButton"
    disableFocusOutline={false}
    role="button"
    tabIndex={-1}
  >
    <span
      className="css-1vry5xk"
      data-cy="badgeButton"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={-1}
    >
      default
    </span>
  </Clickable>
</Memo(BadgeButton)>
`;

exports[`BadgeButton outline 1`] = `
<Memo(BadgeButton)
  appearance="outline"
  onClick={[Function]}
>
  <Clickable
    action={[Function]}
    data-cy="badgeButton"
    disableFocusOutline={false}
    role="button"
    tabIndex={-1}
  >
    <span
      className="css-5i8v04"
      data-cy="badgeButton"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={-1}
    >
      outline
    </span>
  </Clickable>
</Memo(BadgeButton)>
`;

exports[`BadgeButton primary 1`] = `
<Memo(BadgeButton)
  appearance="primary"
  onClick={[Function]}
>
  <Clickable
    action={[Function]}
    data-cy="badgeButton"
    disableFocusOutline={false}
    role="button"
    tabIndex={-1}
  >
    <span
      className="css-1gfuw42"
      data-cy="badgeButton"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={-1}
    >
      primary
    </span>
  </Clickable>
</Memo(BadgeButton)>
`;

exports[`BadgeButton success 1`] = `
<Memo(BadgeButton)
  appearance="success"
  onClick={[Function]}
>
  <Clickable
    action={[Function]}
    data-cy="badgeButton"
    disableFocusOutline={false}
    role="button"
    tabIndex={-1}
  >
    <span
      className="css-92fng0"
      data-cy="badgeButton"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={-1}
    >
      success
    </span>
  </Clickable>
</Memo(BadgeButton)>
`;

exports[`BadgeButton warning 1`] = `
<Memo(BadgeButton)
  appearance="warning"
  onClick={[Function]}
>
  <Clickable
    action={[Function]}
    data-cy="badgeButton"
    disableFocusOutline={false}
    role="button"
    tabIndex={-1}
  >
    <span
      className="css-wso5n9"
      data-cy="badgeButton"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={-1}
    >
      warning
    </span>
  </Clickable>
</Memo(BadgeButton)>
`;
